File: 0559-maximum-depth-of-n-ary-tree/0559-maximum-depth-of-n-ary-tree.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def maxDepth(self, root: 'Node') -> int:
        
        if not root:
            return 0
        
        q = collections.deque([root])
        height = 0
        while q:
            size = len(q)
            for _ in range(size):
                node = q.popleft()
                
                for child in node.children:
                    q.append(child)
            
            height += 1
    
        return height
    

File: 0559-maximum-depth-of-n-ary-tree/0559-maximum-depth-of-n-ary-tree.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    int maxDepth(Node* root) {
        if(!root)   return 0;
        int maxRet = 0;
        for(auto child: root->children){
            maxRet = max(maxRet, maxDepth(child));
        }
        return 1 + maxRet;
    }
};